### DISCUSSION TITLE
Using the executable launcher in Linux, is it possible for me to edit the config somehow?

### DISCUSSION THREAD

**Discussion by wh33t** [LostRuins/koboldcpp#775] [Q&A]
So previously I would install using the installer script that would fetch all the files, then make cublas etc ...

Now I'm giving the launcher a go, I love how simple it is by comparison to start using but I am missing the ability to alter the files that kobold uses for it's interface. Specifically, I like to go in and add an option for auto responses, that allows me to pick, 1,2,3,4,5 # of responses and only with 1 second in between (currently the fastest auto-response time is 15s). My computer isn't very fast, so I will frequently ask the AI a question and then disappear from my machine for a few minutes and it's nice when no time is wasted during this count down and I come back to several replies.

Is there a way for me to alter the files in this way using the stand alone executable?

**Comment by LostRuins** [comment]
Sort of. The klite.embd included is a HTML file containing the code for Kobold Lite (the UI). You can modify that, and then rename it to a `.html` file and open it directly in your browser. It should be able to connect to your koboldcpp instance while preserving any modifications.

Alternatively, Kobold Lite supports User Mods, so you can write a mod to add additional idle timing options and load that mod when you need it.

![image](https://github.com/LostRuins/koboldcpp/assets/39025047/a592d71d-e8b8-4872-acb5-8dcea8c7eea8)

**Reply by alex-ie** [reply]
> Alternatively, Kobold Lite supports User Mods

I could not find them in Lite now,are they gone (2024 year was long ago)?

**Reply by LostRuins** [reply]
They are still there, in Settings > Misc > Advanced.

**Reply by alex-ie** [reply]
@LostRuins , thanks, I have found it, the picture is not near the text now so I missed it initially.

**Comment by wh33t** [comment]
> Sort of. The klite.embd included is a HTML file containing the code for Kobold Lite (the UI). You can modify that, and then rename it to a `.html` file and open it directly in your browser. It should be able to connect to your koboldcpp instance while preserving any modifications.

This klite.embd is a file that exists somewhere after I run the standalone executable? Will it just be in the same directory where I launched the application from?

> Alternatively, Kobold Lite supports User Mods, so you can write a mod to add additional idle timing options and load that mod when you need it.

Neat! What language are the mods written in and would you be so kind as to point me to a place where I can read up on it? I have lots of ideas about how to improve the UI of a KCPP.

**Reply by LostRuins** [reply]
Javascript. There's a sample script when you open it the first time.
